ktlint-rule-engine/src/main/kotlin/com/pinterest/ktlint/rule/engine/internal/SuppressionLocator.kt

```diff
@@ -57,34 +56,27 @@ internal class SuppressionLocator(
     private fun findSuppressionHints(rootNode: ASTNode): List<SuppressionHint> {
         val suppressionHints = ArrayList<SuppressionHint>()
         val commentSuppressionsHints = mutableListOf<CommentSuppressionHint>()
-        rootNode.findSuppressionHints { node ->
-            when (val psi = node.psi) {
-                is PsiComment -> {
+        rootNode.recursiveChildren(includeSelf = true).forEach { node ->
+            val eType = node.elementType
+            when {
+                TokenSets.COMMENTS.contains(eType) -> {
                     node
                         .createSuppressionHintFromComment()
                         ?.let { commentSuppressionsHints.add(it) }
                 }
 
-                is KtAnnotated -> {
-                    psi
-                        .createSuppressionHintFromAnnotations()
-                        ?.let { suppressionHints.add(it) }
+                eType == ElementType.ANNOTATION_ENTRY -> {
+                    createSuppressionHintFromAnnotations(
+                        annotation = node,
+                    )?.let { suppressionHints.add(it) }
                 }
             }
```

Comment:
Lol, I am not a native speaker. 

In Java it was quite common to write checks like `"some text" == someVar` to avoid null pointer exceptions. That also read a bit awkward an not natural, but it was better than writting `someVar != null && someVar == "some text"`. With `someCollection.contains(someVar)` I have that seem feeling as the variable that is being checked comes after the constant. It is probably just a personal preference ;-)